### DISCUSSION TITLE
Stuck on the install/setup stage config file not found

### DISCUSSION THREAD

**Discussion by steven86m** [postalserver/postal#2921] [Help with using Postal]
I try: sudo postal initialize
the reply is: no configuration file provided: not found

I have followed the instructions but they're a bit vague on where I need to save the config file to or point it to?

Any help would be appreciated.

**Comment by willpower232** [comment]
Does it work if you don't use sudo?

**Reply by steven86m** [reply]
No, I get the same message

**Reply by willpower232** [reply]
did you run `postal bootstrap postal.yourdomain.com` before you tried to initialise?

**Reply by steven86m** [reply]
Yeah, I have followed the instructions to this point.

**Reply by steven86m** [reply]
If I re run that command it tells me the latest version number

**Reply by willpower232** [reply]
what are the contents of `/opt/postal/config`?

**Reply by plbeast** [reply]
I got the same problem.  There are three files in /opt/postal/config/
Caddyfile  postal.yml  signing.key
![image](https://github.com/postalserver/postal/assets/3580686/89f44c24-e88e-4dee-90e9-48b544704122)

**Reply by vongohren** [reply]
I had the same issues, at it all related to a docker setup install incorrectly. It worked in the sense that I had docker on the machine and could run docker ps and all. I could even start mariaDB. But something was wrong.

I suggest to just run this script: https://github.com/postalserver/install/blob/main/prerequisites/install-ubuntu.v3.sh,  which is pointed to in the main readme of install: https://github.com/postalserver/install/tree/main

**Comment by steven86m** [comment]
I got a bit further on a fresh install, had to stop due to other work commitments but will pick it up Monday

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by vongohren** [reply]
Did anything happen?

**Comment by plbeast** [comment]
I met the same problem.
And there are three file in /opt/postal/config

![image](https://github.com/postalserver/postal/assets/3580686/a20bb630-56b5-455f-aa58-946d53ba3d24)

**Comment by plbeast** [comment]
I found the COMMAND should excuted in the install path:/opt/postal/install
As the docker-compose.yml is in the direcotry.

**Reply by vongohren** [reply]
Did not do any difference for me

**Reply by lzthqx** [reply]
Has this issue been fixed yet? Following the instructions to the letter simply does not result in a successful install.
Latest Ubuntu, Docker from docker repo, latest git clone etc.
Can someone correct the install instructions, or point me to a resource to correct them myself? Thanks!

**Comment by spoiledtechie** [comment]
Any movement on this or has someone been successful in this?  It has not worked for me and we are getting the same error.

**Reply by vongohren** [reply]
I dont know how I got past it, it just suddenly ende up working. Sorry for the lack of useful information

**Comment by tzalistar** [comment]
I might have found the problem. It seems the bootstrap script creates the file with 640 permission which cause the app inside the container not being able to list the directory or read the files.  I am guissing the app user is postal id:999 which on some distros is mapped to something else. So try playing with the permissions.

What i found that is working for sure is keeping the files in 640 while chaning the owner of the folder and files to 999.

```
chown 999:999 -R config/
chmod  640 -R config/*

ls -lh config/

total 12K
-rw-r-----. 1 systemd-coredump input  371 Feb  5 14:44 Caddyfile
-rw-r-----. 1 systemd-coredump input 1.6K Feb  5 15:00 postal.yml
-rw-r-----. 1 systemd-coredump input  916 Feb  5 14:44 signing.key

```

Hope this helps anyone running to this issue.

**Reply by vongohren** [reply]
Can you consider looking at a PR to change the file creations?

And I wonder why my thing suddenly worked after not doing any of this

**Reply by tzalistar** [reply]
I could yeah, but i want to run some tests first in order find the root cause. One suspect i can think of is security hardening, SELinux, that sort of thing. Perphaps include some sanity checks for config files?

**Reply by vongohren** [reply]
Yeah absolutely. I dont know exactly the config flow and creation of files work so you might know more about that than me

**Comment by plbeast** [comment]
A reinstall and config stritly according the guidelines. and it sucessed.
